platform/logging/layouts/PatternLayout.ts

```diff
@@ -0,0 +1,116 @@
+import { Schema, typeOfSchema } from '../../types';
+import { LogLevel } from '../LogLevel';
+import { LogRecord } from '../LogRecord';
+import { Layout } from './Layouts';
+
+/**
+ * Class that consists of static constants describing supported parameters
+ * in the log message pattern.
+ */
+abstract class Parameters {
```

Comment:
Ha-ha, right, was kind of obsessed with classes at that point :) Do you know if `Object.freeze` gives Typescript a hint that properties are readonly?